app/controllers/api/v4/runs_controller.rb

```diff
@@ -14,11 +14,30 @@ class Api::V4::RunsController < Api::V4::ApplicationController
   end
 
   def show
-    if params[:historic] == '1'
-      @run.parse(fast: false)
-      render json: @run, serializer: Api::V4::RunWithHistorySerializer
+    accept = request.headers.fetch('HTTP_ACCEPT', 'json').downcase
+    if accept == "original_timer"
+      accept = @run.program
+    end
+    timer = Run.program(accept)
+    if accept == 'json' || timer.nil? || !Run.exportable_programs.include?(timer)
```

Comment:
If someone passes in an Accept header we don't understand like `application/xml` or `application/livesolit`, I think it would be clearer to 4XX than to succeed and return JSON.

It looks like the proper way to do this is the [406 error code][1]:
> indicates that the target resource does not have a current representation that would be acceptable to the user agent, according to the proactive negotiation header fields received in the request (Section 5.3), and the server is unwilling to supply a default representation.
>
> The server SHOULD generate a payload containing a list of available representation characteristics and corresponding resource identifiers from which the user or user agent can choose the one most appropriate.

[1]: https://tools.ietf.org/html/rfc7231#section-6.5.6

Reply:
Wasn't sure what the best way to handle bad accept headers were, are there some that should render the default json schema besides `application/json`?

Reply:
Just `application/json` and a missing Accept header should render JSON I think. I can't think of any case other than those two.

Reply:
Would `text/html` be a good one to catch for people testing stuff it it browsers?

Reply:
Ah, I forgot to think about browsers. The endpoint definitely needs to work and return JSON if accessed in a browser. If `text/html` is what browsers send, then yeah let's special case that.